pyanaconda/ui/gui/helpers.py

```diff
@@ -105,6 +104,18 @@ def add_check(self, input_obj, run_check, data=None):
 
         return checkRef
 
+    def can_go_back_focus_if_not(self):
+        """Check whether the input validation checks allow the spoke to be exited.
+
+        Return True if yes, otherwise focus the problematic input field and return False.
+        """
+        failed_check = next(self.failed_checks, None)
+        if failed_check:
+            failed_check.input_obj.grab_focus()
+            return False
+        else:
+            return True
+
```

Comment:
can this function be reused in on_ok_clicked now? (line 166)

Reply:
I would rather keep this separate as one targets spokes and other dialogs.